lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py

```diff
@@ -309,22 +319,38 @@ def __init__(self, module):
         self.entity = None
 
     def get_objects(self):
-        self.datastore = find_datastore_by_name(self.si, self.params['datastore'])
-        if not self.datastore:
-            self.module.fail_json(msg='%(datastore)s could not be located' % self.params)
-
-        self.datacenter = find_datacenter_by_name(self.si, self.params['datacenter'])
+        self.datacenter = self.find_datacenter_by_name(self.params['datacenter'])
         if not self.datacenter:
             self.module.fail_json(msg='%(datacenter)s could not be located' % self.params)
 
+        self.datastore = None
+        datastore_cluster_obj = self.find_datastore_cluster_by_name(self.params['datastore'])
+        if datastore_cluster_obj:
+            datastore = None
+            datastore_freespace = 0
+            for ds in datastore_cluster_obj.childEntity:
+                if isinstance(ds, vim.Datastore) and ds.summary.freeSpace > datastore_freespace:
+                    # If datastore field is provided, filter destination datastores
+                    if ds.summary.maintenanceMode != 'normal':
```

Comment:
what about to check ds.summary.accessible flag here as well?

Reply:
OK. I will add this check as well.